src/main/java/pl/umk/mat/gobooks/common/Audit.java

```diff
@@ -0,0 +1,30 @@
+package pl.umk.mat.gobooks.common;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+
+import javax.persistence.Embeddable;
+import javax.persistence.PrePersist;
+import javax.persistence.PreUpdate;
+import java.time.Instant;
+
+@Embeddable
+@RequiredArgsConstructor
+@Getter @Setter
+public class Audit {
```

Comment:
Lepszym zastosowaniem byłoby użycie adnotacji z hibernate.
Example:
@Column(name = "CreatedDate", updatable=false)
@CreationTimestamp
private Timestamp createdDate;

@Column(name = "ModifiedDate")
@UpdateTimestamp
private Timestamp modifiedDate;

Reply:
Jest to równie dobra metoda, jednak wadą jest zastosowanie adnotacji pochodzących bezpośrednio z Hibernate'a, a nie pakietu `javax.persistence`. W przypadku chęci zmiany dostawcy JPA w trakcie trwania projektu, zaproponowane rozwiązanie wymagałoby przebudowy.